### DISCUSSION TITLE
Distro signatures validation with windows

### DISCUSSION THREAD

**Discussion by tiltingpenguin** [cobbler/cobbler#3462] [General]
We are currently in the process of trying to validate our distro signatures against a JSON schema in libcobblersignatures. However the different Windows distributions are missing most keys that we require from all other distros. I would like to start a discussion here on how we should handle this. From my point of view there are 3 options:

1. We add the required keys for all Windows versions and just leave them empty if we don't need values from them
2. We don't require keys that we don't have in all different distros which would leave us only with "supported arches"
3. We add special rules to the JSON schema to exclude required keys in Windows distributions

Option 2 isn't really practical imo since we want to validate as much of the signatures as possible. 

If you have any other suggestions feel free to add them

**Comment by tiltingpenguin** [comment]
@tpw56j I heard that you are maintaining the Windows feature in cobbler so I would like to have your input on this

**Reply by tpw56j** [reply]
@tiltingpenguin It seems to me that Windows distributions use the same keys in `distro_signatures.json` as Linux distributions. The only exception is `supported_repo_breeds`.

Could you please give examples of other keys that are required by Linux but not used by Windows?

**Reply by tiltingpenguin** [reply]
Sorry about that, I noticed that this was just because some files got out of sync. We fixed this yesterday and now the windows keys are all there. I think we can close this discussion already

**Comment by SchoolGuy** [comment]
IMHO we should add the field to the JSON. This way we keep the schema between the different OS types aligned. When we don't use the field for Windows this shouldn't cause any problems. I doubt that many users will touch the file and if they do, they should use the tool `libcobblersignatures` that we provide.

The problem I have is that then the schema is invalid for versions of Cobbler that are already shipped. That is something that makes me want to adjust the schema in a way that the key is optional.

**Comment by tiltingpenguin** [comment]
I actually just found out that we have the keys in the latest version of the signatures in cobbler.github.io, for some reason they are just missing from the signatures in the 3.x.x folder. Which means this discussion is probably obsolete since we should then add them for version 3.x.x which should for now be the same signatures as the latest one

**Reply by SchoolGuy** [reply]
Did you take into account what file the main repo is shipping?

**Reply by tiltingpenguin** [reply]
Yes, the updated version in cobbler.github.io is based on the recent updates I made to the main repo signatures
